platform.txt

```diff
@@ -5,7 +5,8 @@ version=1.0.0
 
 #compile variables
 ##compile Include 
-compiler.gd.extra_include= "-I{build.source.path}" "-I{build.core.path}/api/deprecated-avr-comp" "-I{build.core.path}/api/deprecated" "-I{build.core.path}/gd32" "-I{build.core.path}/gd32/Source" "-I{build.system.path}/{build.series}_firmware/{build.series}_standard_peripheral/Source" "-I{build.system.path}/{build.series}_firmware/{build.series}_standard_peripheral/Include"  "-I{build.system.path}/CMSIS" "-I{build.system.path}/CMSIS/GD/GD32F30x/Include" "-I{build.system.path}/CMSIS/GD/GD32F30x/Source/GCC" "-I{build.system.path}/CMSIS/GD/GD32F30x/Source"
+compiler.gd.extra_include= "-I{build.source.path}" "-I{build.core.path}/api/deprecated-avr-comp" "-I{build.core.path}/api/deprecated" "-I{build.core.path}/gd32" "-I{build.core.path}/gd32/Source" "-I{build.system.path}/{build.series}_firmware/{build.series}_standard_peripheral/Source" "-I{build.system.path}/{build.series}_firmware/{build.series}_standard_peripheral/Include"  "-I{build.system.path}/{build.series}_firmware/CMSIS" "-I{build.system.path}/{build.series}_firmware/CMSIS/GD/GD32F30x/Include" "-I{build.system.path}/{build.series}_firmware/CMSIS/GD/GD32F30x/Source/GCC" "-I{build.system.path}/{build.series}_firmware/CMSIS/GD/GD32F30x/Source"
```

Comment:
The path change from `-I{build.system.path}/CMSIS` to `-I{build.system.path}/{build.series}_firmware/CMSIS` has to be mirrored in the PlatformIO build script

Reply:
Hm okay so the `CMSIS` folder was moved into the GD32 chip variant folder now, from `system/CMSIS` to `GD32F30x_firmware/CMSIS`. 

The CMSIS header files however should stay the same among all chip series, otherwise we'll have duplicated `core_cm4.h` headers for every Cortex-M4 based GD32 series etc. The CMSIS folder just has the variant-specific e.g. `GD/GD32F30x` folder that can be expanded for other series.

Also all the `CMSIS/GD/GD32F30x` don't make use of the `build.series` so they won't be portable for other series. 

What do you think about changing back the CMSIS folder to its original location?

Reply:
I see what you mean. I think we've got three solutions that should work:
1) Get rid of the `GDF30x_firmware` directory, and move /everything/ up a level. Since `CMSIS` covers all series, and the other directories are already prefixed with their series, this should be sufficient.
2) Just move CMSIS up a level. This would be the fewest changes.
3) Use the current structure. This would prevent any possible conflicts amongst different firmware library imports, as it exactly mirrors the way the libraries are delivered.

Of the three, I'm mostly torn between 1 and 3. The reason I like 1 is that it removes what appears to be a redundant directory, which is always pleasant. The reason I like 3 is that I can see a future where, for whatever reason, the firmware library for one series of chips actually has a different, and incompatible, version of `CMSIS`. Additionally, at least for the F30x series, there are subdirectories of `CMSIS` with the series name embedded in them.

In a perfect world, I'd go with number 1. However, number 3 provides the easiest path going forward for updates in the face of more possibilities, at the cost of redundancy in the source hierarchy. It's less clean, to be sure, but to me, the cost is low compared to my many visions of a future where conflicts occur between firmware library releases for different chip series. It also providies the easiest possible mode of upgrade (just copy the new library in place, no further steps), and compatibility with something like `git subtree` or `git submodule` should that be a desirable future.

Reply:
To add to this, I just did a spot check of https://github.com/CommunityGD32Cores/gigadevice-firmware-and-docs/tree/main/GD32F4xx -- literally the first directory I picked -- to see if there were any differences between revisions:

```sh
ditto:/home…/gigadevice-firmware-and-docs/GD32F4xx% find . -name core_cm4.h | xargs md5sum                                          +-[main]
afc52c631c32e66fed6f4ac859e8d140  ./GD32F403_Demo_Suites_V2.1.0/GD32F403_Firmware_Library/CMSIS/core_cm4.h
afc52c631c32e66fed6f4ac859e8d140  ./GD32F4xx_Demo_Suites_V2.2.0/GD32F4xx_Firmware_Library/CMSIS/core_cm4.h
02df3a9574620f49590e4dacf959cbf3  ./GD32F4xx_Firmware_Library_V1.4/CMSIS/core_cm4.h
afc52c631c32e66fed6f4ac859e8d140  ./GD32F403_Firmware_Library_V2.1.2/Firmware/CMSIS/core_cm4.h
afc52c631c32e66fed6f4ac859e8d140  ./GD32F4xx_Firmware_Library_V2.1.3/Firmware/CMSIS/core_cm4.h
afc52c631c32e66fed6f4ac859e8d140  ./GD32F403_Firmware_Library_V1.2.0/Firmware/CMSIS/core_cm4.h
```

I haven't investigated further, as at this point I'm pretty firmly on the side of just doing the raw copy (option 3 above). This was just to see if one of my complications (where there are differences in what should be generic code) were already true, and it turns out it is from the very first thing I looked at.

Reply:
I see. Then for the sake of easier library imports, let's keep it the current way, that's a strong reason. I'll update the `platform.txt` to use `{build.series}` though and the PlatformIO builder script -- this should not result in any breakage.

Reply:
Adapted in 7a1c54e3dd1574d275c53bd9d3327def1e038371 and 0bb0665f1a8c7c319d3f97942d349d28a2046af0, CI has re-run successfully for all examples in https://github.com/CommunityGD32Cores/gd32-pio-projects/actions/runs/1131498089, I've also re-tested it in the Arduino IDE and it works.